webrtc-c/canary/src/Main.cpp

```diff
@@ -0,0 +1,151 @@
+#include "Include.h"
+
+STATUS onNewConnection(std::shared_ptr<Canary::Peer::Connection>);
+STATUS run(Canary::PConfig);
+VOID sendLocalFrames(Canary::PPeer, MEDIA_STREAM_TRACK_KIND, const std::string&, UINT64, UINT32);
+
+std::atomic<bool> terminated;
+VOID handleSignal(INT32 signal)
+{
+    UNUSED_PARAM(signal);
+    terminated = TRUE;
+}
+
+INT32 main(INT32 argc, CHAR* argv[])
+{
+#ifndef _WIN32
+    signal(SIGINT, handleSignal);
+#endif
+
+    STATUS retStatus = STATUS_SUCCESS;
+    Canary::Config config;
+
+    Aws::SDKOptions options;
+    Aws::InitAPI(options);
+
+    CHK_STATUS(Canary::Config::init(argc, argv, &config));
+    CHK_STATUS(run(&config));
+
+CleanUp:
+    DLOGI("Exiting");
+    Aws::ShutdownAPI(options);
+}
+
+STATUS run(Canary::PConfig pConfig)
+{
+    STATUS retStatus = STATUS_SUCCESS;
+
+    CHK_STATUS(Canary::Cloudwatch::init(pConfig));
+    CHK_STATUS(initKvsWebRtc());
+    SET_LOGGER_LOG_LEVEL(pConfig->logLevel);
+
+    {
+        Canary::Peer::Callbacks callbacks;
+        callbacks.onNewConnection = onNewConnection;
+
+        Canary::Peer peer(pConfig, callbacks);
+
+        CHK_STATUS(peer.init());
+        CHK_STATUS(peer.connect());
+
+        std::thread videoThread(sendLocalFrames, &peer, MEDIA_STREAM_TRACK_KIND_VIDEO, "./assets/h264SampleFrames/frame-%04d.h264",
+                                NUMBER_OF_H264_FRAME_FILES, SAMPLE_VIDEO_FRAME_DURATION);
+        std::thread audioThread(sendLocalFrames, &peer, MEDIA_STREAM_TRACK_KIND_AUDIO, "./assets/opusSampleFrames/sample-%03d.opus",
+                                NUMBER_OF_OPUS_FRAME_FILES, SAMPLE_AUDIO_FRAME_DURATION);
+
+        videoThread.join();
+        audioThread.join();
+        peer.shutdown();
+    }
+
+CleanUp:
+    deinitKvsWebRtc();
+    Canary::Cloudwatch::deinit();
+
+    return retStatus;
+}
+
+STATUS onNewConnection(std::shared_ptr<Canary::Peer::Connection> pConnection)
```

Comment:
Need to think through smart pointers usage and their proliferation.

Reply:
I removed all smart pointers in favor of plain C pointer (except the callbacks from Cloudwatch). This is to make the code simpler, since our SDK is plain C, it's easier to integrate.